FWCore/Framework/interface/stream/callAbilities.h

```diff
@@ -65,33 +65,23 @@ namespace edm {
     using CallGlobal = CallGlobalImpl<T, T::HasAbility::kGlobalCache>;
 
     //********************************
-    // CallGlobalProcessBlock
+    // CallInputProcessBlock
     //********************************
     template <typename T, bool>
-    struct CallGlobalProcessBlockImpl {
-      static void beginProcessBlock(edm::ProcessBlock const& iProcessBlock, typename T::GlobalCache const* iGC) {
-        T::beginProcessBlock(iProcessBlock, iGC);
-      }
-
-      static void accessInputProcessBlock(edm::ProcessBlock const& iProcessBlock, typename T::GlobalCache const* iGC) {
+    struct CallInputProcessBlockImpl {
+      static void accessInputProcessBlock(edm::ProcessBlock const& iProcessBlock, typename T::GlobalCache* iGC) {
```

Comment:
I believe using non-const `T::GlobalCache*` was what we agreed on, correct? My memory is fuzzy.

Reply:
Yes. And the reason was that this function fills the GlobalCache so it is expected that the GlobalCache will change. And also it will be called serially so there is no need for the implementation to make it thread safe.